Src/WindowsServer/WindowsServer.Shared.Tests/FirstChanceExceptionStatisticsTelemetryModuleTest.cs

```diff
@@ -475,6 +478,69 @@ public void FirstChanceExceptionStatisticsTelemetryModuleDoNotIncrementOnRethrow
         }
 
         [TestMethod]
+        public void FirstChanceExceptionStatisticsTelemetryModuleThrowFromTaskAsync()
+        {
+            var metrics = new List<KeyValuePair<Metric, double>>();
+            StubMetricProcessor stub = new StubMetricProcessor()
+            {
+                OnTrack = (m, v) =>
+                {
+                    metrics.Add(new KeyValuePair<Metric, double>(m, v));
+                }
+            };
+
+            using (var module = new FirstChanceExceptionStatisticsTelemetryModule())
+            {
+                module.Initialize(this.configuration);
+                module.MetricManager.MetricProcessors.Add(stub);
+
+                for (int i = 0; i < 10; i++)
+                {
+                    try
+                    {
+                        Task<int> task1 = Task<int>.Factory.StartNew(() => this.Method5(0));
+                        Task<int> task2 = Task<int>.Factory.StartNew(() => this.Method4(0));
+
+                        try
+                        {
+                            task1.Wait();
+                        }
+                        catch (Exception)
+                        {
+                            throw;
+                        }
+
+                        try
+                        {
+                            task2.Wait();
+                        }
+                        catch (Exception)
+                        {
+                            throw;
+                        }
+                    }
+                    catch (Exception)
+                    {
+                    }
+                }
+            }
+
+            Assert.Equal(30, metrics.Count);
```

Comment:
it will also be nice to check that `ExceptionTelemetry` objects will have a different call stacks